napari/utils/colormaps/colormap.py

```diff
@@ -21,41 +22,41 @@ class ColormapInterpolationMode(StringEnum):
     ZERO = auto()
 
 
+@evented_dataclass
 class Colormap:
     """Colormap that relates intensity values to colors.
 
-    Parameters
+    Attributes
     ----------
     colors : array, shape (N, 4)
         Data used in the colormap.
+    name : str
+        Name of the colormap.
     controls : array, shape (N,) or (N+1,)
         Control points of the colormap.
     interpolation : str
         Colormap interpolation mode, either 'linear' or
         'zero'. If 'linear', ncontrols = ncolors (one
         color per control point). If 'zero', ncontrols
         = ncolors+1 (one color per bin).
-    name : str
-        Name of the colormap.
     """
 
-    def __init__(
-        self, colors, *, controls=None, interpolation='linear', name='custom'
-    ):
+    colors: Property[np.ndarray, None, transform_color]
+    name: str = 'custom'
+    controls: np.ndarray = np.zeros((0, 4))
+    interpolation: Property[
+        ColormapInterpolationMode, str, ColormapInterpolationMode
+    ] = ColormapInterpolationMode.LINEAR
 
-        self.name = name
-        self.colors = transform_color(colors)
-        self._interpolation = ColormapInterpolationMode(interpolation)
-        if controls is None:
+    def __post_init__(self):
+        if len(self.controls) == 0:
             n_controls = len(self.colors) + int(
                 self._interpolation == ColormapInterpolationMode.ZERO
             )
             self.controls = np.linspace(0, 1, n_controls)
-        else:
-            self.controls = np.asarray(controls)
```

Comment:
don't want to coerce `self.controls` with `np.asarray` anymore in case something else is passed?

Reply:
yes, will do - good call!

Reply:
done